### PR TITLE
Prevent tackle sound from Queen corpses

### PR DESCRIPTION
**Pull request by Ibessas**
# About the pull request
Fixes Issue #7223

Verifying if the queen is dead on check_bloc to prevent playing the sound.
Tested on almayer with dead queen body, dead drone body ade dead crusher body

# Explain why it's good for the game

Fixing bug to improve game quality

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.
Test: https://www.youtube.com/watch?v=kn_JU-B11pg
</details>

# Changelog
:cl:
fix: Prevent tackle noises from Queen corpses
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
uhh... if another queen evos won't it make the other queen corpse start knocking down again/

**Comment by Ibessas** [comment]
> uhh... if another queen evos won't it make the other queen corpse start knocking down again/

Looking after it

**Comment by ihatethisengine** [comment]
Just check mob's state instead???

queen.stat == DEAD

**Review by realforest2001 — CHANGES_REQUESTED** [review]
As suggested, this needs to be checking if the queen is dead AND unregistering the signal if she is.
`if(is_dead())
	UnregisterSignal(src, COMSIG_MOVABLE_PRE_MOVE)
	return
	`

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 1]
```diff
@@ -443,7 +443,7 @@
 /mob/living/carbon/xenomorph/queen/proc/check_block(mob/queen, turf/new_loc)
 	SIGNAL_HANDLER
```
Comment (realforest2001) [inline code/modules/mob/living/carbon/xenomorph/castes/Queen.dm:444]:
```suggestion
	SIGNAL_HANDLER
	if(is_dead())
		UnregisterSignal(src, COMSIG_MOVABLE_PRE_MOVE)
		return
		
```

File: code/modules/mob/living/carbon/xenomorph/castes/Queen.dm [hunk 2]
```diff
@@ -443,7 +443,7 @@
...
 	for(var/mob/living/carbon/xenomorph/xeno in new_loc.contents)
-		if(xeno.hivenumber == hivenumber)
+		if(xeno.hivenumber == hivenumber && hive.living_xeno_queen)
```
[the remark this answers is not in the document]
Comment (Ibessas) [reply code/modules/mob/living/carbon/xenomorph/castes/Queen.dm]: Done, thanks for the suggestion!

**Review by realforest2001 — APPROVED** [review]

**Comment by harryob** [comment]
by the way, you need to mark your prs as ready to review when you'd like a maintainer to look over and merge

CleanShot 2024-10-26 at 21 21 34@2x

**Review by harryob — CHANGES_REQUESTED** [review]
> What a problem to unreg signal on proc death?

as mentioned, just put the UnregisterSignal line in /mob/living/carbon/xenomorph/queen/death()

**Review by harryob — APPROVED** [review]

**Pull request merged**